packages/web-components/src/components/va-date/va-date.tsx

```diff
@@ -0,0 +1,206 @@
+   */
+  @Prop({ mutable: true }) value: string;
+
+  /**
+   * Fires when the date input loses focus after its value was changed
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  dateChange: EventEmitter;
+
+  /**
+   * Fires when the date input loses focus
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  dateBlur: EventEmitter;
+
+  private handleDateBlur = (event: FocusEvent) => {
+    this.dateBlur.emit(event);
+  };
+
+  private handleDateChange = (event: Event) => {
+    const target = event.target as HTMLSelectElement | HTMLInputElement;
+    let [currentYear, currentMonth, currentDay] = (this.value || '').split('-');
+    if (target.classList.contains('select-month')) {
+      currentMonth = target.value;
+    }
+    if (target.classList.contains('select-day')) {
+      currentDay = target.value;
+    }
+    if (target.classList.contains('input-year')) {
+      currentYear = target.value;
+    }
+    // Use a leading zero for numbers < 10
+    const numFormatter = new Intl.NumberFormat('en-US', {
+      minimumIntegerDigits: 2,
+    });
+
+    // If Month or Date Select are chosen as empty return ''
+    // Otherwise convert numbers less than 10 to have a leading 0
+    this.value = `${currentYear}-${
+      currentMonth ? numFormatter.format(parseInt(currentMonth)) : ''
+    }-${currentDay ? numFormatter.format(parseInt(currentDay)) : ''}`;
+
+    // This event should always fire to allow for validation handling
+    this.dateChange.emit(event);
+  };
+
+  /**
+   * Whether or not an analytics event will be fired.
+   */
+  @Prop() enableAnalytics: boolean;
+
+  /**
+   * The event used to track usage of the component. This is emitted when an
+   * input value changes and enableAnalytics is true.
+   */
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  render() {
+    const {
+      required,
+      label,
+      name,
+      error,
+      maxYear,
+      minYear,
+      handleDateBlur,
+      handleDateChange,
+      value,
+    } = this;
+
+    // Convert string to number to remove leading 0 on values less than 10
+    const [year, month, day] = (value || '')
+      .split('-')
+      .map(val => parseInt(val, 10));
+
+    const daysForSelectedMonth = month > 0 ? days[month] : [];
+
+    // Check validity of date if invalid provide message and error state styling
+    const dateInvalid =
+      required && (!isFullDate(value) || day > daysForSelectedMonth.length)
+        ? 'Please provide a valid date'
+        : null;
+
+    // Setting new attribute to avoid conflicts with only using error attribute
+    // Error attribute should be leveraged for custom error messaging
+    // Fieldset has an implicit aria role of group
+    return (
+      <Host value={value} invalid={dateInvalid}>
+        <fieldset aria-label="Select Month and two digit day XX and four digit year format XXX">
```

Comment:
@k80bowman we'll need to make sure we get a translation for this as well since this is new.

Reply:
Good call, I'll add it to the spreadsheet and alert the check-in team. Thanks!

Reply:
We're also [missing an X](https://github.com/department-of-veterans-affairs/va.gov-team/issues/39047#issuecomment-1104468838).

```suggestion
        <fieldset aria-label="Select Month and two digit day XX and four digit year format XXXX">
```